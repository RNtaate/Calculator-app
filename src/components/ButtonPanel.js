import React from 'react';
import Button from './Button';

let ButtonPanel = () => { 

  let displayButton = (myName) => {
    return(
      <Button name={myName}/>
    )
  }

  return(
    <div className='calculator-panel'>

      <div className='group1'>
        {displayButton('AC')}
        {displayButton('+/-')}
        {displayButton('%')}
        {displayButton('รท')}
      </div>
      <div className='group2'>
        {displayButton('7')}
        {displayButton('8')}
        {displayButton('9')}
        {displayButton('X')}
      </div>   
      <div className='group3'>
        {displayButton('4')}
        {displayButton('5')}
        {displayButton('6')}
        {displayButton('-')}
      </div>   
      <div className='group4'>
        {displayButton('1')}
        {displayButton('2')}
        {displayButton('3')}
        {displayButton('+')}
      </div>  
      <div className='group5'>
        {displayButton('0')}
        {displayButton('.')}
        {displayButton('=')}
      </div>      

    </div>
  )
}

export default ButtonPanel;