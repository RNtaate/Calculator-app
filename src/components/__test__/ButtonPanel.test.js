import React from 'react';
import ReactDOM from 'react-dom';
import ButtonPanel from '../ButtonPanel';

import renderer from 'react-test-renderer';

const handle = () => {}; 

it('renders Navbar component without crashing', () => {
  const myDiv = document.createElement('div');
  ReactDOM.render(<ButtonPanel clickHandler={handle}/>, myDiv);
});

it('matches the Button panel component snapshot', () => {
  const tree = renderer.create(<ButtonPanel clickHandler={handle}/>);
  expect(tree).toMatchSnapshot();
});