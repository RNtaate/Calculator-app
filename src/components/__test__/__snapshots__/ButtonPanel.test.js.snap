// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the Button panel component snapshot 1`] = `
<div
  className="calculator-panel"
>
  <div
    className="group1"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="%"
    >
      %
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "width": "25%",
        }
      }
      type="button"
      value="รท"
    >
      รท
    </button>
  </div>
  <div
    className="group2"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="7"
    >
      7
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="8"
    >
      8
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="9"
    >
      9
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "width": "25%",
        }
      }
      type="button"
      value="X"
    >
      X
    </button>
  </div>
  <div
    className="group3"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="4"
    >
      4
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="5"
    >
      5
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="6"
    >
      6
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "width": "25%",
        }
      }
      type="button"
      value="-"
    >
      -
    </button>
  </div>
  <div
    className="group4"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="1"
    >
      1
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="2"
    >
      2
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="3"
    >
      3
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "width": "25%",
        }
      }
      type="button"
      value="+"
    >
      +
    </button>
  </div>
  <div
    className="group5"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "50%",
        }
      }
      type="button"
      value="0"
    >
      0
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "width": "25%",
        }
      }
      type="button"
      value="."
    >
      .
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "width": "25%",
        }
      }
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
